dll/ntdll/rtl/version.c

```diff
@@ -52,20 +53,28 @@ SetRosSpecificInfo(IN OUT PRTL_OSVERSIONINFOEXW VersionInformation)
             (kvpInfo->Type == REG_DWORD) &&
             (kvpInfo->DataLength == sizeof(ULONG)))
         {
-            /* Is the value set? */
-            ReportAsWorkstation = *(PULONG)kvpInfo->Data;
-            if ((VersionInformation->wProductType == VER_NT_SERVER) &&
-                (ReportAsWorkstation != 0))
+            ULONG IsWorkstation = SharedUserData->NtProductType == NtProductWinNt;
+            ULONG ReportAsWorkstation = (*(PULONG)kvpInfo->Data) != 0;
+            if (IsWorkstation != ReportAsWorkstation)
             {
-                /* It is, modify the product type to report a workstation */
-                VersionInformation->wProductType = VER_NT_WORKSTATION;
-                DPRINT("We modified the reported OS from NtProductServer to NtProductWinNt\n");
+                g_ReportProductType = ReportAsWorkstation ? NtProductWinNt : NtProductServer;
```

Comment:
VER_NT_WORKSTATION and VER_NT_SERVER
(since the value in g_ ... goes into the wProductType. Same numerical values but different types.)